src/Go/effects/netscriptGoImplementation.ts

```diff
@@ -150,8 +158,8 @@ export async function getOpponentNextMove(logOpponentMove = true, logger: (s: st
 /**
  * Returns a grid of booleans indicating if the coordinates at that location are a valid move for the player (black pieces)
  */
-export function getValidMoves() {
-  const boardState = Go.currentGame;
+export function getValidMoves(_boardState?: SimpleBoard) {
+  const boardState = _boardState ? getNewBoardStateFromSimpleBoard(_boardState) : Go.currentGame;
```

Comment:
The fact that you have to transform this into a BoardState, unlike the others, hints at the underlying issue with giving this an optional parameter of a SimpleBoard: Determining valid moves requires knowing if certain moves are illegal due to ko, and this simply cannot be represented from a SimpleBoard representation. Given the subtlety of this particular foot-gun, I think this one would be better to omit (unless you can come up with a way to provide a full BoardState as input.)